### DISCUSSION TITLE
How to let AREPL detect libraries installed with PDM

### DISCUSSION THREAD

**Discussion by jamestansx** [Almenon/AREPL-vscode#419] [Q&A] [answered]
[pdm](https://github.com/pdm-project/pdm) installs libraries into the local project directory `__pypackage__`. However, when running AREPL, it doesn't detect libraries in the folder. 

How to setup so that AREPL can detect the libraries?

**Comment by Almenon** [comment, marked as answer]
Oh god there's another python package management tool? 😱 Python really needs to settle on one of them. 

Anyways, unfortunately AREPL doesn't have a setting to recognize the ` __pypackage__` folder automatically. You can manually specify it with the following code:

```python
import sys
import os
sys.path.insert(0,os.path.abspath('__pypackages__\\3.8-32\\lib'))
```

Replace `3.8-32` with your python version. If you're on a mac/linux OS use `/` instead of `\\`

You can test if it works like so:

```python
import sys
import os
sys.path.insert(0,os.path.abspath('__pypackages__\\3.8-32\\lib'))

import <yourLibrary>
print(<yourLibrary>.__version__) # the version number should match the one installed in pdm
```

If you want to you can dive into the AREPL code itself (do `print(sys.path[-1])` in AREPL to see the location) and add it in so you don't have to specify it.

For example, I added `sys.path.insert(0,os.path.abspath('__pypackages__\\3.8-32\\lib'))` into `C:\Users\caleb\.vscode\extensions\almenon.arepl-2.0.3\node_modules\arepl-backend\python\arepl_python_evaluator.py`, right before line 242 (run_context.run). Although note that when the next update comes the code will be overwritten. It will probably be a long time before the next update.

**Reply by jamestansx** [reply]
Thanks for the answer! 


> Oh god there's another python package management tool? 😱 Python really needs to settle on one of them.

Ya, agree. Personally, I think PDM is quite nice.
